src/pages/teams/[teamId]/dashboard/residents/[residentId]/index.tsx

```diff
@@ -39,6 +46,44 @@ const ResidentPage: NextPage<WithUser<ResidentPageProps>> = ({
     (ds) => ds.state == 'REJECTED'
   );
 
+  interface EvidenceAwaitingSubmission {
+    documentType: string;
+    dateRequested: DateTime | undefined;
+    requestedBy: string | undefined;
+  }
+  const findEvidenceAwaitingSubmissions = () => {
```

Comment:
If this is an expensive thing to do then look into the `useMemo` React hook so that it won't be recalculated unless wether `evidenceRequests` or `documentSubmissions` changes.

Reply:
Addressed